src/test/java/com/bakdata/kafka/util/SchemaTopicClientTest.java

```diff
@@ -47,64 +47,142 @@
                 .with(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class)
-                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, this.schemaRegistryMockExtension.getUrl())
+                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                        this.schemaRegistryMockExtension.getUrl())
                 .build();
         this.kafkaCluster.send(sendRequest);
 
         final SchemaRegistryClient client = this.schemaRegistryMockExtension.getSchemaRegistryClient();
-        assertThat(client.getAllSubjects())
+        this.softly.assertThat(client.getAllSubjects())
                 .contains(TOPIC + "-value");
 
-        try (final SchemaTopicClient schemaTopicClient = this.createSchemaTopicClient()) {
+        try (final SchemaTopicClient schemaTopicClient = this.createClientWithSchemaRegistry()) {
             schemaTopicClient.deleteTopicAndResetSchemaRegistry(TOPIC);
         }
 
         delay(TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        assertThat(this.kafkaCluster.exists(TOPIC))
-                .as("Topic is deleted")
+
+        this.softly.assertThat(client.getAllSubjects())
+                .doesNotContain(TOPIC + "-value");
+        this.softly.assertThat(this.kafkaCluster.exists(TOPIC))
                 .isFalse();
-        assertThat(client.getAllSubjects())
+    }
+
+    @Test
+    void shouldResetSchema() throws InterruptedException, IOException, RestClientException {
+        this.kafkaCluster.createTopic(TopicConfig.withName(TOPIC).useDefaults());
+        this.softly.assertThat(this.kafkaCluster.exists(TOPIC))
+                .as("Topic is created")
+                .isTrue();
+
+        final SendValuesTransactional<TestRecord> sendRequest = SendValuesTransactional
+                .inTransaction(TOPIC, List.of(TestRecord.newBuilder().setContent("foo").build()))
+                .with(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class)
+                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                        this.schemaRegistryMockExtension.getUrl())
+                .build();
+        this.kafkaCluster.send(sendRequest);
+
+        final SchemaRegistryClient client = this.schemaRegistryMockExtension.getSchemaRegistryClient();
+        this.softly.assertThat(client.getAllSubjects())
+                .contains(TOPIC + "-value");
+
+        try (final SchemaTopicClient schemaTopicClient = this.createClientWithSchemaRegistry()) {
+            schemaTopicClient.resetSchemaRegistry(TOPIC);
+        }
+
+        delay(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        this.softly.assertThat(client.getAllSubjects())
                 .doesNotContain(TOPIC + "-value");
+        this.softly.assertThat(this.kafkaCluster.exists(TOPIC))
+                .isTrue();
+    }
+
+    @Test
+    void shouldDeleteTopicAndKeepSchemaWhenSchemaRegistryUrlIsNotSet() throws InterruptedException, RestClientException,
```

Comment:
This currently results in a failure and I think we should try to keep it that way, because I'd argue this is always an oversight and a misconfiguration

Reply:
Maybe we can check if any serializer or deserializer is an instance of AbstractKafkaSchemaSerDe and then require the SR url to be set?

Reply:
What is resulting in a failure? Tests are passing, no?

Reply:
By currently I mean the current version of stream-bootstrap without the changes of this PR. I think `shouldDeleteTopicAndKeepSchemaWhenSchemaRegistryUrlIsNotSet` is not a behavior we ever want

Reply:
But that is what this PR is about. Clean was failing because SR was not configured (or rather was configured but there was no endpoint to call). That's why we changed clean up behavior and only call SR if it is configured. Otherwise it will be logged: https://github.com/bakdata/streams-bootstrap/pull/187/files#diff-7b49d6ef833e64cf10ebb2f829f2444c33a61923e58889c9d34c0a2cb4fc64a8R113. Because you usually call clean with the same values as deploy, SR should be configured consistently

Reply:
I mean in this case the user also wouldn't be able to deploy because SR is not configured and the AvroSerde does not work. So I think it is fine. Also is there a way to check all configured Serdes in Kafka Streams?

Reply:
@philipp94831 maybe something like this could work:
```java
    public final Properties getKafkaProperties() {
        final Properties kafkaConfig = this.createKafkaProperties();

        EnvironmentStreamsConfigParser.parseVariables(System.getenv())
                .forEach(kafkaConfig::setProperty);
        this.streamsConfig.forEach(kafkaConfig::setProperty);

        final StreamsConfig config = new StreamsConfig(kafkaConfig);

        if ((config.defaultValueSerde() instanceof SpecificAvroSerde
                || config.defaultKeySerde() instanceof SpecificAvroSerde) && this.schemaRegistryUrl == null) {
            System.out.println("Oh! Bad config! Watch out!");
        }

        return kafkaConfig;
    }
```

Reply:
You can still manually set serdes for Stores/Topics/...

Reply:
I can remove this test if it is ambiguous.

Reply:
I've looked into it and it seems to me there is no way for us to get Serdes from a topology.